File: spec/javascript/test_enum_invalid.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('EnumInvalid', 'src/term_strz.bin', function(r, EnumInvalid) {
    assert.equal(r.pet1, EnumInvalid.Animal.DOG);
    assert.equal(r.pet2, 0x6f);
});

File: spec/python/test_enum_invalid.py
import unittest

from enum_invalid import EnumInvalid

class TestEnumInvalid(unittest.TestCase):
    def test_enum_invalid(self):
        with EnumInvalid.from_file("src/term_strz.bin") as r:
            self.assertEqual(r.pet_1, EnumInvalid.Animal.dog)
            self.assertEqual(r.pet_2, 0x6f)
